test/StorageRent/StorageRent.t.sol

```diff
@@ -0,0 +1,1312 @@
+            })
+        );
+
+        vm.expectRevert(StorageRent.StaleAnswer.selector);
+        vm.prank(admin);
+        fcStorage.refreshPrice();
+    }
+
+    function testPriceFeedRevertsIncompleteRound() public {
+        // Set zero timeStamp value
+        priceFeed.setRoundData(
+            MockChainlinkFeed.RoundData({
+                roundId: 1,
+                answer: 2000e8,
+                startedAt: block.timestamp,
+                timeStamp: 0,
+                answeredInRound: 1
+            })
+        );
+        vm.expectRevert(StorageRent.IncompleteRound.selector);
+        vm.prank(admin);
+        fcStorage.refreshPrice();
+    }
+
+    function testUptimeFeedRevertsSequencerDown(int256 answer) public {
+        if (answer == 0) ++answer;
+        // Set nonzero answer. It's counterintuitive, but a zero answer
+        // means the sequencer is up.
+        uptimeFeed.setRoundData(
+            MockChainlinkFeed.RoundData({
+                roundId: 1,
+                answer: answer,
+                startedAt: 0,
+                timeStamp: block.timestamp,
+                answeredInRound: 1
+            })
+        );
+
+        vm.expectRevert(StorageRent.SequencerDown.selector);
+        vm.prank(admin);
+        fcStorage.refreshPrice();
+    }
+
+    function testUptimeFeedRevertsStaleAnswer() public {
+        // Set stale answeredInRound value
+        uptimeFeed.setRoundData(
+            MockChainlinkFeed.RoundData({
+                roundId: 2,
+                answer: 0,
+                startedAt: block.timestamp,
+                timeStamp: block.timestamp,
+                answeredInRound: 1
+            })
+        );
+
+        vm.expectRevert(StorageRent.StaleAnswer.selector);
+        vm.prank(admin);
+        fcStorage.refreshPrice();
+    }
+
+    function testUptimeFeedRevertsIncompleteRound() public {
+        // Set zero timeStamp value
+        uptimeFeed.setRoundData(
+            MockChainlinkFeed.RoundData({
+                roundId: 1,
+                answer: 0,
+                startedAt: block.timestamp,
+                timeStamp: 0,
+                answeredInRound: 1
+            })
+        );
+        vm.expectRevert(StorageRent.IncompleteRound.selector);
+        vm.prank(admin);
+        fcStorage.refreshPrice();
+    }
+
+    function testUptimeFeedRevertsGracePeriodNotOver() public {
+        // Set startedAt == timeStamp, meaning the sequencer just restarted.
+        uptimeFeed.setRoundData(
+            MockChainlinkFeed.RoundData({
+                roundId: 1,
+                answer: 0,
+                startedAt: block.timestamp,
+                timeStamp: block.timestamp,
+                answeredInRound: 1
+            })
+        );
+
+        vm.expectRevert(StorageRent.GracePeriodNotOver.selector);
+        vm.prank(admin);
+        fcStorage.refreshPrice();
+    }
+
+    function testFuzzOnlyAuthorizedCanRefreshPrice(address caller) public {
```

Comment:
do we ever explicitly test that the admin can call this function? i know we do it implicitly in a bunch of places, but wondering if we should have an explicit check like we do with FuzzCredit